eng/build.sh

```diff
@@ -367,6 +371,7 @@ if [ "$build_repo_tasks" = true ]; then
         -p:Configuration=Release \
         -p:Restore=$run_restore \
         -p:Build=true \
+        -bl:RepoTasks.binlog \
```

Comment:
This was added in a patch so for completeness sake I've added this here. I don't think we actually want to take this change though.

Reply:
Yup, I tried for about 5 minutes before I realized I'm too unfamiliar with bash.

Reply:
> I don't think we actually want to take this change though.

Agree we shouldn't create binary logs unless they were requested on the command line. The spot to change would be https://github.com/dotnet/aspnetcore/pull/32790/files#diff-cf7b73895c3c9b73ee94c6e628c9948b2efc248f3c715e3744348c621deda69cR342-R344 Idea would be to update `$toolset_build_args` as well as `$msbuild_args` i.e. add a line similar to https://github.com/dotnet/aspnetcore/blob/main/eng/build.ps1#L393 and an `else` block w/ something like https://github.com/dotnet/aspnetcore/blob/main/eng/build.ps1#L400.

Your choice whether to fix things now but I suggest at least undoing this change.

Reply:
Updated to honour binarylog option